pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -371,8 +374,6 @@ func WithEksaRegion(eksaRegion string) func(client *PackageControllerClient) {
 	return func(config *PackageControllerClient) {
 		if eksaRegion != "" {
 			config.eksaRegion = eksaRegion
-		} else {
-			config.eksaRegion = eksaDefaultRegion
```

Comment:
Isn't this already setting a default value for this field?

Reply:
It doesn't, no.

If a user calls `NewPackageControllerClient` without any options, the `eksaRegion` is just `""`. 

In order to get the benefit of this code fragment, one has to call `NewPackageControllerClient(..., curatedpackages.WithEksaRegion(""))`

Why would someone ever call `curatedpackages.WithEksaRegion` with an empty value? That's very unintuitive to me.

Reply:
The intention of the WithEksaRegion is to allow the reading of configs from environment variables: [factory.go](https://github.com/aws/eks-anywhere/blob/d4e1f9cd05761dc16dfd25709609623bb511a3ef/pkg/dependencies/factory.go#L1000) then passing this forward to the WithEksaRegion to determine whether to use the default region or not. So the issue here is not calling with an empty value but rather determining from the environment variable whether or not to set to default. The only use cases I am currently aware of this functionality is through the factory.go by the creation of packagecontrollerclient. Is there another use case I am not aware of here?

Reply:
Yes. Full cluster lifecycle doesn't have the ability to use environment variables. 

All the same, I don't see any reason this isn't an improvement? Does it break the desired functionality?